cvmfs/cvmfs_server

```diff
@@ -5624,6 +5650,47 @@ migrate_2_1_20() {
   load_repo_config $name
 }
 
+migrate_2_2_0() {
+  local name=$1
+  local destination_version="2.3.0-1"
+  local server_conf="/etc/cvmfs/repositories.d/${name}/server.conf"
+
+  # get repository information
+  load_repo_config $name
+  [ ! -z $CVMFS_SPOOL_DIR     ] || die "\$CVMFS_SPOOL_DIR is not set"
+  [ ! -z $CVMFS_USER          ] || die "\$CVMFS_USER is not set"
+  [ ! -z $CVMFS_UNION_FS_TYPE ] || die "\$CVMFS_UNION_FS_TYPE is not set"
+
+  echo "--> umount repository"
+  run_suid_helper rw_umount $name
+
+  echo "--> updating scratch directory layout"
```

Comment:
Potentially... We could check for open transactions in `migrate()` but that is going to cause a dead-lock. `cvmfs_server` will prevent you from closing the transaction with `cvmfs_server abort` if the version number doesn't fit.

Anyway: That might be more favourable than deleting the scratch directory which might contain valuable data.